packages/frontend/src/components/Modal/AccountDetailModal.tsx

```diff
@@ -0,0 +1,113 @@
+import Jazzicon from '@metamask/jazzicon'
+import { shortenAddress, useEthers } from '@usedapp/core'
+import copyToClipboard from 'copy-to-clipboard'
+import { useEffect, useRef } from 'react'
+import { CopyIcon, RedirectIcon } from 'src/components//Icons'
+import { Button } from 'src/components/Buttons/Button'
+import { ContentRow, Modal } from 'src/components/Modal/Modal'
+import { useChainId } from 'src/hooks/chainId/useChainId'
+import { Colors } from 'src/styles/colors'
+import { getExplorerAddressLink } from 'src/utils/getExplorerLink'
+import styled, { css } from 'styled-components'
+
+export interface ModalProps {
+  isShown: boolean | undefined
+  onRequestClose: () => void
+  wallet?: string
+}
+
+export const AccountDetailModal = ({ isShown, onRequestClose, wallet = 'Metamask' }: ModalProps) => {
+  const { account, deactivate } = useEthers()
+  const accountIconRef = useRef<any>(null)
+  const chainId = useChainId()
+
+  useEffect(() => {
+    if (account && accountIconRef.current) {
+      accountIconRef.current.innerHTML = ''
+      accountIconRef.current.appendChild(Jazzicon(40, parseInt(account.slice(2, 10), 16)))
+    }
+  }, [account])
+
+  return (
+    <Modal isShown={isShown} onRequestClose={onRequestClose} title="Your account">
+      <ContentWrapper>
+        <ContentRow>
+          <ConnectedWallet style={{ margin: 0 }}>Connected with {wallet}</ConnectedWallet>
+        </ContentRow>
+        <ContentRow>
+          <AccountIcon ref={accountIconRef} />
+          <AccountAddress>{shortenAddress(account || '')}</AccountAddress>
+        </ContentRow>
+        {account && (
+          <ContentRow>
+            <AccountLink
+              href={account && getExplorerAddressLink(chainId, account)}
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              <RedirectIcon size={24} color={Colors.Blue} />
+              View in block explorer
+            </AccountLink>
```

Comment:
Clicking "View in block explorer" opens a new tab with localhost:3000 for me.

Reply:
Links work on testnet, but not on hardhat.
There is a screenshot from netlify: 
<img width="742" alt="Screenshot 2022-05-13 at 06 32 20" src="https://user-images.githubusercontent.com/66270386/168211677-927cf134-4c84-408e-a164-9d67920c8795.png">